### PR TITLE
Signing maven pushes with PGP

### PR DESCRIPTION
**Pull request by shs96c**

### CODE REVIEW

**Comment by shs96c** [comment]
Please land PR #977 first!

File: src/com/facebook/buck/maven/Publisher.java [hunk 1]
```diff
@@ -246,15 +273,65 @@ public DeployResult publish(List<Artifact> toPublish) throws DeploymentException
...
+          .addCommand(
+              "gpg",
+              "-ab",
+              "--batch",
+              "--passphrase", signingPassphrase.get(),
+              file.getAbsolutePath())
+          .build();
+      ProcessExecutor.Result result = processExecutor.launchAndExecute(args);
+      if (result.getExitCode() != 0) {
+        throw new HumanReadableException("Unable to sign %s", file);
```
Comment (illicitonion) [inline src/com/facebook/buck/maven/Publisher.java]: Actually include the stdout/stderr here.

**Comment by illicitonion** [comment]
Throw in a check in PublisherIntegrationTest#testPublishFiles that we get the .asc and I'm happy.

File: test/com/facebook/buck/maven/PublisherIntegrationTest.java [hunk 2]
```diff
@@ -87,4 +108,67 @@ public void testPublishFiles() throws Exception {
...
+          }
+        } catch (IOException e) {
+          throw new RuntimeException(e);
+        }
+        return executor;
+      }
+    };
+
+    publisher.publish(ImmutableSet.of(publishable));
+  }
```
Comment (illicitonion) [inline test/com/facebook/buck/maven/PublisherIntegrationTest.java]: This is missing the assert that the .asc file was actually PUT?

Comment (shs96c) [inline test/com/facebook/buck/maven/PublisherIntegrationTest.java]: If the `.asc` is missing and signing has been enabled, the check at `Publisher.java:322` will cause an exception to be thrown. The test will fail if the executor isn't called, and I've added an assert that every expected command is executed.

File: test/com/facebook/buck/maven/PublisherIntegrationTest.java [hunk 3]
```diff
@@ -87,4 +108,75 @@ public void testPublishFiles() throws Exception {
...
+          throw new RuntimeException(e);
+        }
+        return executor;
+      }
+    };
+
+    publisher.publish(ImmutableSet.of(publishable));
+
+    for (ProcessExecutorParams params : commands.keySet()) {
+      executor.isProcessLaunched(params);
```
Comment (illicitonion) [inline test/com/facebook/buck/maven/PublisherIntegrationTest.java]: Actually assert on this random boolean value that you're just computing? :)

Comment (shs96c) [inline test/com/facebook/buck/maven/PublisherIntegrationTest.java]: Hahaha! I thought the method made the assert already. *facepalm*

**Comment by v-jizhang** [comment]
Sorry we never got back to this! Thank you for your contributions. At this point, we are closing this because it is quite out of date and it is not clear if there is still interest in this feature. If you are still interested in adding PGP signing support, please submit a new PR and we promise to look at it in less than 3 years. Thank you (and sorry) again.

**Pull request closed without merging**